homeassistant/components/fritzbox_callmonitor/config_flow.py

```diff
@@ -0,0 +1,130 @@
+"""Config flow for fritzbox_callmonitor."""
+
+from fritzconnection.core.exceptions import FritzConnectionException
+import requests
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
+
+from .base import FritzBoxPhonebook
+
+# pylint:disable=unused-import
+from .const import (
+    CONF_PHONEBOOK,
+    CONF_PREFIXES,
+    DEFAULT_HOST,
+    DEFAULT_PHONEBOOK,
+    DEFAULT_PORT,
+    DEFAULT_USERNAME,
+    DOMAIN,
+)
+
+DATA_SCHEMA_USER = vol.Schema(
+    {
+        vol.Required(CONF_HOST, default=DEFAULT_HOST): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): vol.Coerce(int),
+        vol.Required(CONF_USERNAME, default=DEFAULT_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Optional(CONF_PHONEBOOK, default=DEFAULT_PHONEBOOK): int,
```

Comment:
Can we query how many phonebooks there are? And can we not show this field if there is only 1. 

Would also be nice if we can show the phonebook names (if they have those)?

Reply:
Yes, we can query how many phonebooks are stored on the FritzBox. That is a great idea to hide this field if only one phonebook exists.

Was considering the same thing in the last couple days and I am pretty sure that once a connection is established the names can be queried as well. Therefore, I will see what I can do. May have to rewrite the base class in the process.

Reply:
Done in https://github.com/home-assistant/core/pull/40736/commits/f88f633a12142c9f6af148368270efdc13e3d9f7:

- skip option if only one phonebook is available
- show drop down of available phonebooks' names